### PR TITLE
Feat/cw2 info

### PR DESCRIPTION
**Pull request by bkioshn**
## Describe your changes
1. Add CW2 info column to code table including
- code list (my stored and my saved code)
- Recent code
- Public code
2. Code details
- Upper section
4. Contract details 
- Migration table

Note: if there is no info, will display "N/A" 
5. Remove unnecessary type in public project page

### CODE REVIEW

File: src/lib/pages/code-details/index.tsx [hunk 1]
```diff
@@ -56,10 +58,22 @@ const CodeDetailsBody = observer(
...
+            <Flex gap={2}>
+              <Text fontWeight={500} color="text.dark" variant="body2">
+                CW2 Info:
+              </Text>
+              <Text
+                color={cw2Info ? "text.main" : "text.disabled"}
+                variant="body2"
+                wordBreak="break-all"
+              >
+                {cw2Info || "N/A"}
```
Comment (evilpeach) [inline src/lib/pages/code-details/index.tsx:74]: `cw2Info ?? "N/A"`

File: src/lib/pages/codes/components/CodesTable.tsx [hunk 2]
```diff
@@ -144,6 +147,14 @@ const CodeTableRow = ({ code }: CodesRowProps) => {
       <TableRowNoBorder>
         <CodeNameCell code={code} />
       </TableRowNoBorder>
+      <TableRowNoBorder>
+        <Text
+          color={cw2Info ? "text.main" : "text.disabled"}
+          wordBreak="break-all"
+        >
+          {cw2Info || "N/A"}
```
Comment (evilpeach) [inline src/lib/pages/codes/components/CodesTable.tsx:155]: ditto

File: src/lib/pages/contract-details/components/tables/migration/MigrationRow.tsx [hunk 3]
```diff
@@ -76,6 +78,14 @@ export const MigrationRow = ({
           }}
         />
       </TableRow>
+      <TableRow>
+        <Text
+          color={cw2Info ? "text.main" : "text.disabled"}
+          wordBreak="break-all"
+        >
+          {cw2Info || "N/A"}
```
Comment (evilpeach) [inline src/lib/pages/contract-details/components/tables/migration/MigrationRow.tsx:86]: ditto

File: src/lib/pages/contract-details/components/tables/migration/index.tsx [hunk 4]
```diff
@@ -66,10 +66,10 @@ export const MigrationTable = ({
     );
 
   const templateColumns =
-    "90px minmax(300px, 1fr) repeat(2, max(150px)) max(232px) max(180px)";
+    "90px minmax(300px, 1fr) minmax(220px, 1fr) repeat(2, max(150px)) max(232px) max(180px)";
 
   return (
-    <TableContainer overflow="visible">
+    <TableContainer>
```
Comment (evilpeach) [inline src/lib/pages/contract-details/components/tables/migration/index.tsx:72]: Will it break the old interface?

File: src/lib/pages/public-project/components/table/code/PublicProjectCodeRow.tsx [hunk 5]
```diff
@@ -21,13 +22,15 @@ export const PublicProjectCodeRow = ({
 }: CodeTableRowProps) => {
   const navigate = useInternalNavigate();
   const { currentChainName } = useWallet();
-
+  const { publicInfo } = publicCodeInfo;
```
Comment (evilpeach) [inline src/lib/pages/public-project/components/table/code/PublicProjectCodeRow.tsx:25]: On line 20, you can destructure as follows.
```tsx
publicCodeInfo: { publicInfo },
```
and remove line 25

File: src/lib/pages/public-project/components/table/code/PublicProjectCodeRow.tsx [hunk 6]
```diff
@@ -41,46 +44,48 @@ export const PublicProjectCodeRow = ({
...
-        <Text>{publicCodeInfo.publicInfo.name}</Text>
+        <Text>{publicInfo.name}</Text>
+      </TableRowNoBorder>
+      <TableRowNoBorder>
+        <Text
+          color={cw2Info ? "text.main" : "text.disabled"}
+          wordBreak="break-all"
+          whiteSpace="pre-wrap"
+        >
+          {cw2Info || "N/A"}
```
Comment (evilpeach) [inline src/lib/pages/public-project/components/table/code/PublicProjectCodeRow.tsx:61]: use `??`

**Review by evilpeach — CHANGES_REQUESTED** [review]

File: src/lib/pages/contract-details/components/tables/migration/index.tsx (hunk 4 above)
Comment (bkioshn) [reply src/lib/pages/contract-details/components/tables/migration/index.tsx:72]: Nope

**Review by evilpeach — APPROVED** [review]
lgtm

**Pull request merged**